File: generate-keys.js
/**
 * Génère une paire de clés RSA-2048 pour les démonstrations JWT.
 * Les clés sont sauvegardées dans ./keys/ (non committées dans Git).
 *
 * Usage : node generate-keys.js
 */

const crypto = require('crypto');
const fs     = require('fs');
const path   = require('path');

const keysDir = path.join(__dirname, 'keys');
if (!fs.existsSync(keysDir)) fs.mkdirSync(keysDir);

console.log('Génération de la paire de clés RSA-2048...');

const { privateKey, publicKey } = crypto.generateKeyPairSync('rsa', {
  modulusLength: 2048,
  publicKeyEncoding:  { type: 'spki',  format: 'pem' },
  privateKeyEncoding: { type: 'pkcs8', format: 'pem' },
});

fs.writeFileSync(path.join(keysDir, 'private.pem'), privateKey);
fs.writeFileSync(path.join(keysDir, 'public.pem'),  publicKey);

console.log('✓ keys/private.pem  (GARDEZ SECRÈTE)');
console.log('✓ keys/public.pem   (exposée par /api/public-key)');

File: generate_keys.py
import os
from cryptography.hazmat.primitives.asymmetric import rsa
from cryptography.hazmat.primitives import serialization

keys_dir = os.path.join(os.path.dirname(__file__), "keys")
os.makedirs(keys_dir, exist_ok=True)

print("Generation de la paire de cles RSA-2048...")

private_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)

private_pem = private_key.private_bytes(
    encoding=serialization.Encoding.PEM,
    format=serialization.PrivateFormat.PKCS8,
    encryption_algorithm=serialization.NoEncryption(),
)

public_pem = private_key.public_key().public_bytes(
    encoding=serialization.Encoding.PEM,
    format=serialization.PublicFormat.SubjectPublicKeyInfo,
)

with open(os.path.join(keys_dir, "private.pem"), "wb") as f:
    f.write(private_pem)

with open(os.path.join(keys_dir, "public.pem"), "wb") as f:
    f.write(public_pem)

print("keys/private.pem")
print("keys/public.pem")
